### PR TITLE
alice-tools: init at 0.12.1

### PR DESCRIPTION
**Pull request by OPNA2608**
###### Description of changes

Packages [alice-tools](https://github.com/nunuhara/alice-tools), a collection of tools for viewing and converting file formats used in AliceSoft games.

image

I have checked CLI/Qt5/Qt6 variants on x86_64-linux & aarch64-linux, and CLI/Qt5 variants on x86_64-darwin. My macOS version is seemingly too old for Qt6 so trying to use those binaries just errors on my end. I have checked the builds by opening the graphics-containing archive from one of my games, browsing through its contents and extracting some of the assets into png & webp formats.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/games/alice-tools/default.nix
Comment (OPNA2608) [inline pkgs/tools/games/alice-tools/default.nix:59]: With Qt5 detection works just fine:
```
Run-time dependency qt5 (modules: Core, Gui, Widgets) found: YES 5.15.7 (pkg-config)
Detecting Qt5 tools
Run-time dependency qt5 (modules: Core) found: YES 5.15.7 (pkg-config)
Program /nix/store/1dcavvraw4q6xxs2pkiwyviv2g80nqd7-qtbase-5.15.7-dev/bin/moc found: YES 5.15.7 (/nix/store/1dcavvraw4q6xxs2pkiwyviv2g80nqd7-qtbase-5.15.7-dev/bin/moc)
Program /nix/store/1dcavvraw4q6xxs2pkiwyviv2g80nqd7-qtbase-5.15.7-dev/bin/uic found: YES 5.15.7 (/nix/store/1dcavvraw4q6xxs2pkiwyviv2g80nqd7-qtbase-5.15.7-dev/bin/uic)
Program /nix/store/1dcavvraw4q6xxs2pkiwyviv2g80nqd7-qtbase-5.15.7-dev/bin/rcc found: YES 5.15.7 (/nix/store/1dcavvraw4q6xxs2pkiwyviv2g80nqd7-qtbase-5.15.7-dev/bin/rcc)
Program /nix/store/1dcavvraw4q6xxs2pkiwyviv2g80nqd7-qtbase-5.15.7-dev/bin/lrelease found: NO
Program lrelease5 found: NO
Program lrelease-qt5 found: NO
Program lrelease found: YES 5.15.7 (/nix/store/fs86kagqgwn6iabm13hxh0fifxgyfw6c-qttools-5.15.7-dev/bin/lrelease)
```

With Qt6, it doesn't:
```
Run-time dependency qt6 (modules: Core, Gui, Widgets) found: YES 6.4.1 (pkg-config)
Detecting Qt6 tools
Run-time dependency qt6 (modules: Core) found: YES 6.4.1 (pkg-config)
Program /nix/store/sgx4h5qca3y6qx6cpj4ssycmr5kw9vn1-qtbase-6.4.1-dev/bin/moc found: NO
Program /nix/store/j4rg2wb4z7kh7a8ciqcp20p69kr346wv-qtbase-6.4.1/./libexec/moc found: NO
Program moc6 found: NO
Program moc-qt6 found: NO
Program moc found: NO
Program /nix/store/sgx4h5qca3y6qx6cpj4ssycmr5kw9vn1-qtbase-6.4.1-dev/bin/uic found: NO
Program /nix/store/j4rg2wb4z7kh7a8ciqcp20p69kr346wv-qtbase-6.4.1/./libexec/uic found: NO
Program uic6 found: NO
Program uic-qt6 found: NO
Program uic found: NO
Program /nix/store/sgx4h5qca3y6qx6cpj4ssycmr5kw9vn1-qtbase-6.4.1-dev/bin/rcc found: NO
Program /nix/store/j4rg2wb4z7kh7a8ciqcp20p69kr346wv-qtbase-6.4.1/./libexec/rcc found: NO
Program rcc6 found: NO
Program rcc-qt6 found: NO
Program rcc found: NO
Program /nix/store/sgx4h5qca3y6qx6cpj4ssycmr5kw9vn1-qtbase-6.4.1-dev/bin/lrelease found: NO
Program /nix/store/j4rg2wb4z7kh7a8ciqcp20p69kr346wv-qtbase-6.4.1/./libexec/lrelease found: NO
Program lrelease6 found: NO
Program lrelease-qt6 found: NO
Program lrelease found: YES 6.4.1 (/nix/store/0bjv6cn2qr4lg9j13gxydi7n65qipyzf-qttools-6.4.1-dev/bin/lrelease)

src/meson.build:129:4: ERROR: MOC sources specified and couldn't find uic-qt6, please check your qt6 installation
```

In our Qt6 package `moc`, `uic` and `rrc` are in `${qtbase.dev}/libexec` while in Qt5 they were in `${qtbase.dev}/bin`. Meson looks in `${qtbase.dev}/bin`, `${qtbase}/libexec` and `$PATH` before giving up. Not sure if the bins are supposed to be in `${qtbase.dev}/bin` or the Meson Qt6 detection needs to look in  `${qtbase.dev}/libexec` instead.

**Comment by OPNA2608** [comment]
@ofborg build alice-tools libsForQt5.alice-tools qt6Packages.alice-tools

**Comment by OPNA2608** [comment]
Hmm looks like the default stdenv's compiler on both Darwin platforms is also too old for Qt6, throws the same "Qt requires a C++17 compiler" error on Ofborg…

File: pkgs/tools/games/alice-tools/default.nix
Comment (lilyinstarlight) [inline pkgs/tools/games/alice-tools/default.nix:37]: Remove one or both of these? We probably should clarify if this comment extends to both patches

**Review by lilyinstarlight — APPROVED** [review]
Aside from the one nit and the one mentioned Qt issue (which is something that needs to be fixed for Qt's pkg-config files in a separate PR), this looks fabulous to me

File: pkgs/tools/games/alice-tools/default.nix
Comment (OPNA2608) [reply pkgs/tools/games/alice-tools/default.nix:37]: One patch is for the `libsys4` submodule which handles the format parsing & converting, the other for `alice-tools` which provides the CLI/GUI and calls into the library. Both need a patch for it to work. Will clarify with a comment.

Comment (OPNA2608) [reply pkgs/tools/games/alice-tools/default.nix]: Edited the comment so it refers to both patches.

**Review by lilyinstarlight — APPROVED** [review]

File: pkgs/tools/games/alice-tools/default.nix
Comment (lilyinstarlight) [reply pkgs/tools/games/alice-tools/default.nix:56]: I've opened #210134 to fix this. Once this PR is merged, I'll open another PR to undo the hack in staging

Comment (lilyinstarlight) [reply pkgs/tools/games/alice-tools/default.nix:56]: (Building this PR with the hack removed but on top of that other PR does in fact build -- so yeah this seems to be the fix)

Comment (lilyinstarlight) [reply pkgs/tools/games/alice-tools/default.nix:56]: The fix for this just hit master. Can you see if it works against master without this hack now?

Comment (OPNA2608) [reply pkgs/tools/games/alice-tools/default.nix:56]: Yep works fine now, will update.

Comment (lilyinstarlight) [reply pkgs/tools/games/alice-tools/default.nix]: Sweet, if ofborg builds then this PR looks fabulous to me so I'll approve and then see if we can get a committer to review/merge it

**Review by lilyinstarlight — APPROVED** [review]

**Pull request merged**
